File: 08/03/PageObjects/LoginPage.cs
using OpenQA.Selenium;
using NUnit.Framework;

namespace PageObjects
{
    class LoginPage
    {
        IWebDriver Driver;
        By LoginForm = By.Id("login");
        By UsernameInput = By.Id("username");
        By PasswordInput = By.Id("password");
        By SubmitButton = By.CssSelector("button");
        By SuccessMessage = By.CssSelector(".flash.success");
        By FailureMessage = By.CssSelector(".flash.error");

        public LoginPage(IWebDriver Driver)
        {
            this.Driver = Driver;
            Driver.Navigate().GoToUrl("http://the-internet.herokuapp.com/login");
            Assert.That(Driver.FindElement(LoginForm).Displayed);
        }

        public void With(string Username, string Password)
        {
            Driver.FindElement(UsernameInput).SendKeys(Username);
            Driver.FindElement(PasswordInput).SendKeys(Password);
            Driver.FindElement(SubmitButton).Click();
        }

        public bool SuccessMessagePresent()
        {
            return Driver.FindElement(SuccessMessage).Displayed;
        }

        public bool FailureMessagePresent()
        {
            return Driver.FindElement(FailureMessage).Displayed;
        }
    }
}

File: python/08/03/pages/login_page.py
from selenium.webdriver.common.by import By


class LoginPage():
    _login_form = {"by": By.ID, "value": "login"}
    _username_input = {"by": By.ID, "value": "username"}
    _password_input = {"by": By.ID, "value": "password"}
    _submit_button = {"by": By.CSS_SELECTOR, "value": "button"}
    _success_message = {"by": By.CSS_SELECTOR, "value": ".flash.success"}
    _failure_message = {"by": By.CSS_SELECTOR, "value": ".flash.error"}

    def __init__(self, driver):
        self.driver = driver
        self.driver.get("http://the-internet.herokuapp.com/login")
        assert self.driver.find_element(
            self._login_form["by"], self._login_form["value"]).is_displayed()

    def with_(self, username, password):
        self.driver.find_element(self._username_input["by"],
                                 self._username_input["value"]).send_keys(username)
        self.driver.find_element(self._password_input["by"],
                                 self._password_input["value"]).send_keys(password)
        self.driver.find_element(self._submit_button["by"],
                                 self._submit_button["value"]).click()

    def success_message_present(self):
        return self.driver.find_element(
            self._success_message["by"], self._success_message["value"]).is_displayed()

    def failure_message_present(self):
        return self.driver.find_element(
            self._failure_message["by"], self._failure_message["value"]).is_displayed()
